codec-sockjs/src/main/java/io/netty/handler/codec/sockjs/transport/HttpResponseBuilder.java

```diff
@@ -0,0 +1,353 @@
+    public HttpResponseBuilder header(final AsciiString name, AsciiString value) {
+        headers.add(name.toString(), value.toString());
+        return this;
+    }
+
+    /**
+     * Adds a header to this HTTP response.
+     *
+     * @param name the name of the HTTP header.
+     * @param values the values of hte HTTP header.
+     * @return {@link HttpResponseBuilder} to support method chaining.
+     */
+    public HttpResponseBuilder header(final CharSequence name, final Iterable<? extends CharSequence> values) {
+        headers.add(name, values);
+        return this;
+    }
+
+    /**
+     * Sets the HTTP header {@link HttpHeaderNames#SET_COOKIE} if {@link SockJsConfig#areCookiesNeeded()}
+     * is true.
+     *
+     * @param config the {@link SockJsConfig}
+     * @return {@link HttpResponseBuilder} to support method chaining.
+     */
+    public HttpResponseBuilder setCookie(final SockJsConfig config) {
+        if (config.areCookiesNeeded()) {
+            header(HttpHeaderNames.SET_COOKIE, encodeSessionIdCookie(request));
+        }
+        return this;
+    }
+
+    /**
+     * Builds a {@link HttpResponse} response using the configuration options
+     * previsouly set.
+     *
+     * @return {@link HttpResponse} the configured HTTP response.
+     */
+    public HttpResponse buildResponse() {
+        final HttpResponse response = new DefaultHttpResponse(request.protocolVersion(), status);
+        addAllHeaders(response);
+        return response;
+    }
+
+    /**
+     * Builds a {@link FullHttpResponse} response using the passed-in {@link ByteBufAllocator} when
+     * creating the response body.
+     *
+     * @return {@link FullHttpResponse} the configured HTTP response.
+     */
+    public FullHttpResponse buildFullResponse() {
+        if (byteBuf == null && content == null) {
+            throw new IllegalArgumentException("Content must be supplied to build a FullHttpResponse");
+        }
+        final ByteBuf buf = byteBuf == null ? copiedBuffer(content, UTF_8) : byteBuf;
+        final DefaultFullHttpResponse response = new DefaultFullHttpResponse(request.protocolVersion(), status, buf);
+        header(HttpHeaderNames.CONTENT_LENGTH, String.valueOf(buf.readableBytes()));
+        addAllHeaders(response);
+        return response;
+    }
+
+    /**
+     * Creates a builder for the passed in {@link HttpRequest}.
+     *
+     * @param request the {@link HttpRequest} for which a response should be built.
+     * @return {@link HttpResponseBuilder} the builder instance to be further configured.
+     */
+    public static HttpResponseBuilder responseFor(final HttpRequest request) {
+        return new HttpResponseBuilder(request);
+    }
+
+    private void addAllHeaders(final HttpResponse response) {
+        if (contentType != null) {
+            response.headers().set(CONTENT_TYPE, contentType);
+        }
+        if (chunked) {
+            if (response.protocolVersion() == HttpVersion.HTTP_1_1) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1066)